plugin/core/rpc.py

```diff
@@ -173,14 +176,15 @@ def on_notification(self, notification_method: str, handler: 'Callable') -> None
         self._notification_handlers[notification_method] = handler
 
     def request_handler(self, request: 'Dict[str, Any]') -> None:
+        request_id = request.get("id")
         params = request.get("params", dict())
         method = request.get("method", '')
         debug('<--  ' + method)
         if self.settings.log_payloads and params:
             debug('     ' + str(params))
         if method in self._request_handlers:
             try:
-                self._request_handlers[method](params)
+                self._request_handlers[method](params, request_id)
```

Comment:
Would adding this extra request_id argument not break the other request handlers? e.g. workspace/applyEdit?